pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -570,7 +570,12 @@ func getNodeSecurityGroupIDForLB(compute *gophercloud.ServiceClient, nodes []*v1
 		// case 2: node1:SG1,SG2  node2:SG3,SG4  return SG1,SG3
 		// case 3: node1:SG1,SG2  node2:SG2,SG3  return SG1,SG2
 		securityGroupName := srv.SecurityGroups[0]["name"]
-		nodeSecurityGroupIDs.Insert(securityGroupName.(string))
+		secGroupID, err := groups.IDFromName(network, securityGroupName.(string))
```

Comment:
`groups.IDFromName` does an API lookup.  Given that almost all uses of this loop are going to be performing many lookups for a very small (almost always 1) number of names, it would be nice to do this lookup _after_ de-duplication...

iow: something like:

Assuming `s/nodeSecurityGroupIDs/sgNames/g`

```golang
for _, node := range nodes {
   ...
   sgNames.Insert(securityGroupName.(string))
}

ids := make([]string, sgNames.Len())
for i, name := range sgNames.List() {
    var err error
    ids[i], err = groups.IDFromName(network, name)
    if err != nil {
        return []string{}, err
    }
}

return ids, nil
```

Reply:
Yep, thanks. Fix applied. (force push -- is that OK?)

What do you think about turning the function into a method?